README.md
# CMDYTPD_v3.1.2(farewell)

CMDYTPD v3.x_beta
by Dee


Donate in <a href='https://www.paypal.me/wfrdee'>Paypal</a><br>
***************  ChangeLog   ***************<br>
***************Version (3.1.2)**************<br><br>
#Final Version<br>
	Final Version I dont know when to continue to develop this app
	<br>
***************Version: (3.1)***************

#Bug Fixes<br>
	About Section Bug Fixed!<br>
	Choosing a search result fix<br>

***************Version: (3.0)***************

#Optimized<br>
	(3.0) Code Optimization and changes(2.2_unreleased)<br><br>

#Online Radio<br>
	(3.0) Added Online Radio<br><br>

#Bug fix<br>
	(3.0) OTA app update fix and improvements(2.2_unreleased)<br>
	(3.0) Updater app update(2.2_unreleased)<br><br>

***************Version: (2.1)***************<br>

#Playlist<br>
	(2.1) You can now Play or Download YouTube Playlists<br><br>

#Help Section Added<br>
	(2.1) Help Section is now Updated!<br><br>

#Search History Added<br>
	(2.1) Search History Added!<br><br>

#App OTA Update Added<br>
	(2.1) App OTA update is now added for easy and fast application update in just one click!<br><br>

#Optimized!<br>
	(2.1) Code improvements and optimized!<br><br>

#Bug Fixed!<br>
	(2.1) Fixed! Search Result Bug<br>
	(2.1) Fixed! Exits after playing.<br><br>


***************Version: (2.0)***************<br><br>

#Search Menu Added!<br>
	(2.0) You can now Search youtube videoes and stream or download<br><br>

#Optimized!<br>
	(2.0) Optimized and Much Faster..<br><br>


***************Version: (2.0)***************<br><br>

#You can download and stream music on youtube<br>
	(1.0) You can now download or stream music using youtube url.<br><br>

#More Bugs To Fix Later..<br>






history.txt
fuck this shit
hello
third
second
first


main.h

/*


CMD YouTube Player and Downloader v2.1_beta

Release Update

(2.2)
1. Bug Fix

(2.1)
1. Playlist Player
2. Search History
3. Help Section Updated
4. App OTA update
5. App Optimization
6. Bug Fixed

(2.0)
1. Youtube Search added
2. App Optimization


*/
////////






#include <iostream>
#include <conio.h>
#include "proccess_lib.h"
using namespace std;


int main(){
	system(("title CMDYTPD " + get_version()).c_str());
	updater_update();
/////////////////// Code Testing Area  ///////////////////////////////
//	cout << "Testing Code...." << endl;
//
//	ifstream file("history.txt");
//	string line = "";
//	history_count = 0;
//	while(file >> line){
//		cout << history_count << line << endl;
//		//search_history[history_count] = line;
//		//cout << search_history[history_count] << " = " << line << endl;
//		history_count++;
//	}
//
//
//	get_update();
//	cout << "\\" << endl;
//
//
//
//
//	system("pause");
////////////////////////////////////////////////////////////////////////
	int ch;
	string search = "";
	while(ch != 8){
		title_display();
		cout << "1. Search\n2. Download\n3. Play\n4. Online Radio\n5. Help\n6. About\n7. Check for Update\n8. Exit\nCommand: ";
		cin >> ch;
		switch(ch){
			case 1: {
				title_display();
				cout << "Search Query: " << endl;
				get_history(true);
				string search_command = "";
				char character = ' ';
				while(character != 13){
					if(kbhit()){
						character = getch();
						if((character == 8) && (search.length() > 0)){
							search.erase(search.length()-1);
						}
						else{
							search += character;
							string first_command = "lib\\youtube-dl ytsearch20:\"";
							string second_command = "\" --get-title --get-id";
							search_command = first_command;
							search_command += search;
							search_command += second_command;
						}
						title_display();
						cout << "Search Query: " ;
						cout << search << endl;
						get_history(true);
//						cout << search_command;
					}
				}
				cs();
				add_history(search);
				search_youtube(search_command, search);
				break;
			}
			case 2: {
				title_display();
				cout << "Download URL: " ;
				string url;
				cin >> url;
				title_display();
				cout << "1. Download Video\n2. Download Audio\n\nCommand: ";
				int ch1;
				cin >> ch1;
				switch(ch1){
					case 1:{
						download_video(url);
						break;
					}
					case 2:{
						download_audio(url);
						break;
					}
				}
				break;
			}
			case 3: {
				title_display();
				cout << "Stream URL: " ;
				string url;
				cin >> url;
				title_display();
				cout << "By Dee" << endl << endl;
				cout << "1. Stream Video\n2. Stream Audio\n\nCommand: ";
				int ch1;
				cin >> ch1;
				switch(ch1){
					case 1:{
						stream_video(url);
						break;
					}
					case 2:{
						stream_audio(url);
						break;
					}
				}
				break;
			}
			case 4: {
				get_radio();
				break;
			}
			case 5: {
				title_display();
				cout << "Help" << endl << endl;
				cout << "Key\t\t\tDescription" << endl;
				cout << "'space' or 'p'\t\tPause or Play" << endl;
				cout << "'q'\t\t\tEnd music" << endl;
				cout << "'arrow keys'\t\tSeek" << endl;
				cout << "'m'\t\t\tMute" << endl;
				cout << "'j'\t\t\tSelect Subtitle" << endl;
				cout << "'v'\t\t\tSubtitle On and Off" << endl;
				cout << "'['']'\t\t\tChange Speed" << endl;
				cout << "'9''0'\t\t\tVolume" << endl;
				cout << endl;
				system("pause");
				break;
			}
			case 6: {
				title_display();
				cout << "Copyright 2018-2019" << endl;
				cout << "\nThanks to" << endl;
				cout << "Github" << endl;
				cout << "YouTube" << endl;
				cout << "Google" << endl;
				cout << "mpv Player" << endl;
				cout << "StackOverFlow" << endl;
				cout << "" << endl;
				cout << "This App is For Personal Use Only Not For Sale" << endl;
				cout << "Download Copyright Musics at your own RISK!" << endl;
				cout << "Dont Blame me if you are in jail. :)" << endl;
				cout << "" << endl;
				cout << "About (CMDYTPD) v3.1_beta" << endl;
				cout << "" << endl;
				cout << "1. Bug Fixes\n\t(3.1) About Section Bug Fixed!\n\t(3.1) Choosing a search result fix" << endl;
				cout << "About (CMDYTPD) v3.0_beta" << endl;
				cout << "" << endl;
				cout << "1. Optimized\n\t(3.0) Code Optimization and changes(2.2_unreleased)" << endl;
				cout << "2. Online Radio\n\t(3.0) Added Online Radio" << endl;
				cout << "3. Bug fix\n\t(3.0) OTA app update fix and improvements(2.2_unreleased)\n\t(3.0) Updater app update(2.2_unreleased)" << endl << endl;
				cout << "About (CMDYTPD) v2.1_beta" << endl;
				cout << "" << endl;
				cout << "1. Playlist\n\t(2.1) You can now Play or Download YouTube Playlists" << endl;
				cout << "2. Help Section Added\n\t(2.1) Help Section is now Updated!" << endl;
				cout << "3. Search History Added\n\t(2.1) Search History Added!" << endl;
				cout << "4. App OTA Update Added\n\t(2.1) App OTA update is now added for easy and fast application update in just one click!" << endl;
				cout << "5. Optimized!\n\t(2.1) Code improvements and optimized!" << endl;
				cout << "6. Bug Fixed!\n\t(2.1) Fixed! Search Result Bug\n\t(2.1) Fixed! Exits after playing." << endl << endl;
				cout << "About (CMDYTPD) v2.0_beta" << endl;
				cout << "" << endl;
				cout << "1. Search Menu Added!\n\t(2.0) You can now Search youtube videoes and stream or download" << endl;
				cout << "2. Optimized!\n\t(2.0) Optimized and Much Faster.." << endl;
				cout << "About (CMDYTPD) v1.0_alpha" << endl;
				cout << "" << endl;
				cout << "1. You can download and stream music on youtube\n\t(1.0) You can now download or stream music using youtube url." << endl;
				cout << "2. More Bugs To Fix Later.." << endl << endl;
				system("pause");
				break;
			}
			case 7:{
			    get_update();
				return 0;
				break;
			}
			case 8 :{
				return 0;
				break;
			}
		}

	}


}


main_lib.cpp

/*


CMD YouTube Player and Downloader v2.1_beta

Release Update
(2.1)
1. Playlist Player
2. Search History
3. Help Section Updated
4. App Optimization
5. Bug Fixed

(2.0)
1. Youtube Search added
2. App Optimization


*/






#include <iostream>
#include <conio.h>
#include "proccess_lib.h"
using namespace std;


int main(){
	system(("title CMDYTPD " + get_version()).c_str());
/////////////////// Code Testing Area  //////////////////////////
//	cout << "Testing Code...." << endl;
//	
//	ifstream file("history.txt");
//	string line = "";
//	history_count = 0;
//	while(file >> line){
//		cout << history_count << line << endl;
//		//search_history[history_count] = line;
//		//cout << search_history[history_count] << " = " << line << endl;
//		history_count++;
//	}
//
//
//
//
//
//
//
//
//	system("pause");
////////////////////////////////////////////////////////////////
	int ch;
	string search = "";
	while(ch != 6){
		title_display();
		cout << "1. Search\n2. Download\n3. Play\n4. Help\n5. About\n6. Exit\nCommand: ";
		cin >> ch;
		switch(ch){
			case 1: {
				title_display();
				cout << "Search Query: " << endl;
				get_history(true);
				string search_command = "";
				char character = ' ';
				while(character != 13){
					if(kbhit()){
						character = getch();
						if((character == 8) && (search.length() > 0)){
							search.erase(search.length()-1);
						}
						else{
							search += character;
							string first_command = "youtube-dl ytsearch20:\"";
							string second_command = "\" --get-title --get-id";
							search_command = first_command;
							search_command += search;
							search_command += second_command;
						}
						title_display();
						cout << "Search Query: " ;
						cout << search << endl;
						get_history(true);
//						cout << search_command;
					}
				}
				cs();
				add_history(search);
				search_youtube(search_command, search);
				break;
			}
			case 2: {
				title_display();
				cout << "Download URL: " ;
				string url;
				cin >> url;
				title_display();
				cout << "1. Download Video\n2. Download Audio\n\nCommand: ";
				int ch1;
				cin >> ch1;
				switch(ch1){
					case 1:{
						download_video(url);
						break;
					}
					case 2:{
						download_audio(url);
						break;
					}
				}
				break;
			}
			case 3: {
				title_display();
				cout << "Stream URL: " ;
				string url;
				cin >> url;
				title_display();
				cout << "By Dee" << endl << endl;
				cout << "1. Stream Video\n2. Stream Audio\n\nCommand: ";
				int ch1;
				cin >> ch1;
				switch(ch1){
					case 1:{
						stream_video(url);
						break;
					}
					case 2:{
						stream_audio(url);
						break;
					}
				}
				break;
			}
			case 4: {
				title_display();
				cout << "Help" << endl << endl;
				cout << "Key\t\t\tDescription" << endl;
				cout << "'space' or 'p'\t\tPause or Play" << endl;
				cout << "'q'\t\t\tEnd music" << endl;
				cout << "'arrow keys'\t\tSeek" << endl;
				cout << "'m'\t\t\tMute" << endl;
				cout << "'j'\t\t\tSelect Subtitle" << endl;
				cout << "'v'\t\t\tSubtitle On and Off" << endl;
				cout << "'['']'\t\t\tChange Speed" << endl;
				cout << "'9''0'\t\t\tVolume" << endl;
				cout << endl;
				system("pause");
				break;
			}
			case 5: {
				title_display();
				cout << "Copyright 2018-2019" << endl;
				cout << "\nThanks to" << endl;
				cout << "Github" << endl;
				cout << "YouTube" << endl;
				cout << "Google" << endl;
				cout << "mpv Player" << endl;
				cout << "StackOverFlow" << endl;
				cout << "" << endl;
				cout << "This App is For Personal Use Only Not For Sale" << endl;
				cout << "Download Copyright Musics at your own RISK!" << endl;
				cout << "Dont Blame me if you are in jail. :)" << endl;
				cout << "" << endl;
				cout << "About (CMDYTPD) v2.1_beta" << endl;
				cout << "" << endl;
				cout << "1. Playlist\n\t(2.1) You can now Play or Download YouTube Playlists" << endl;
				cout << "2. Help Section Added\n\t(2.1) Help Section is now Updated!" << endl;
				cout << "3. Search History Added\n\t(2.1) Search History Added!" << endl;
				cout << "4. Optimized!\n\t(2.1) Code improvements and optimized!" << endl;
				cout << "5. Bug Fixed!\n\t(2.1) Fixed! Search Result Bug\n\t(2.1) Fixed! Exits after playing." << endl;
				cout << "About (CMDYTPD) v2.0_beta" << endl;
				cout << "" << endl;
				cout << "1. Search Menu Added!\n\t(2.0) You can now Search youtube videoes and stream or download" << endl;
				cout << "2. Optimized!\n\t(2.0) Optimized and Much Faster.." << endl;
				cout << "About (CMDYTPD) v1.0_alpha" << endl;
				cout << "" << endl;
				cout << "1. You can download and stream music on youtube\n\t(1.0) You can now download or stream music using youtube url." << endl;
				cout << "2. More Bugs To Fix Later.." << endl;
				system("pause");
				break;
			}
			case 6 :{
				return 0;
				break;
			}
			case 7:{
				search_results();
				break;
			}
		}
		
	}
	
	
}


proccess_lib.h
/*
Memo


youtube-dl gvsearch5:how to develop for android --no-playlist --write-info-json --write-annotation --write-thumbnail --write-sub --skip-download


*/

#include <iostream>
#include <fstream>
#include <string>
#include <windows.h>


using namespace std;

// Objects
void cs();
void download_video(string video_url);
void download_audio(string audio_url);
void stream_video(string video_url);
void stream_audio(string audio_url);
int main();

// Universal Variable
string search_result[20][2] = {};
string search_history[10] = {};
int history_count = 0;


string get_version(){
	return "v3.1.2_beta";
}

void title_display(){

	cs();
	cout << "CMD YouTube Player and Downloader(CMDYTPD) " << get_version() << endl;
	cout << "By Dee" << endl << endl;

}

void get_result(){

	ifstream file("lib\\result.txt");
	string line = "";
	int counter = 0;
	int counter1 = 0;
    while (getline (file,line)){
		if(counter % 2 == 0){
			search_result[counter1][0] = line;
			cout << counter1+1 << ". " << line << endl;
		}
		else{
			search_result[counter1][1] = line;
			counter1++;
		}
		counter++;
    }
    file.close();

}

/// Radio Online List
// https://www.dropbox.com/s/5q89kilper41jbt/radio_link.txt?dl=1

void play_radio(string name, string link){

	title_display();
	cout << "Playing: " << name << endl;
	system(("lib\\mpv " + link +" --no-video").c_str());
	system("pause");

}


void get_radio(){
	title_display();
	cout << "Getting Online Radio Channels..." << endl;
	string command1 = "powershell -Command \"(New-Object Net.WebClient).DownloadFile('https://www.dropbox.com/s/5q89kilper41jbt/radio_link.txt?dl=1', 'lib\\radio_link.txt')\"";
	string command2 = "powershell -Command \"Invoke-WebRequest https://www.dropbox.com/s/5q89kilper41jbt/radio_link.txt?dl=1 -OutFile lib\\radio_link.txt\"";
	system((command1 + "\n" + command2).c_str());
	title_display();
	cout << "Available Radio Channel" << endl;
	ifstream file("lib\\radio_link.txt");
	string temp1, temp2;
	string links[100][2] = {{}};
	int i = 0;
	while(file >> temp1 >> temp2){
		links[i][0] = temp1;
		links[i][1] = temp2;
		cout << i+1 << ". " << links[i][0] << endl;
		i++;
	}
	cout << "Command: ";
	int ch;
	cin >> ch;
	play_radio(links[ch-1][0], links[ch-1][1]);

}




void get_history(bool display){
	ifstream file("lib\\history.txt");
	string line = "";
	history_count = 0;
	while(file >> line && history_count < 20){
		search_history[history_count] = line;
		history_count++;
	}
	file.close();
	if(display == true){
		cout << "Search History" << endl;
		for(int i = 0; i < 20; i++){
			if(search_history[i].empty()){
				break;
			}
			cout << i+1 << ". " << search_history[i] << endl;
		}
	}
}

void add_history(string search){

	string temp_data[100] = {};
	temp_data[0] = search;
	for(int i = 0; i <= history_count; i++){
		temp_data[i+1] = search_history[i];
	}
	ofstream file("lib\\history.txt");
	for(int i = 0; i <= history_count; i++){
		file << temp_data[i] << endl;
	}
	file.close();
}


std::string removeAll( std::string str, const std::string& from) {
    size_t start_pos = 0;
    while( ( start_pos = str.find( from)) != std::string::npos) {
        str.erase( start_pos, from.length());
    }
    return str;
}

//void search_display(string s){
//	while(1){
//		title_display();
//		cout << "Search Input: " << s << endl;
//		cout << "Loading";
//		for(int i= 0; i < 5; i++){
//			Sleep(500);
//			cout << ".";
//		}
//	}
////	return NULL;
//}

//void start_search(string search_command){
//	system((search_command + " > " + "lib\\result.txt").c_str());
//
//}

string search_youtube(string search_command, string search){
	int ss = 1;
	string line = "";
	int result_counter = 0;
	//system("pause");
	title_display();
	cout << "Search Input: " << search  << endl;
	cout << "Loading.." << endl;
	system((search_command + " > " + "lib\\result.txt").c_str());
    cout << "Search Results" << endl;
    get_result();
    cout << "Command: ";
    int choice;
    cin >> choice;
	title_display();
	choice--;
	cout << "Choice: " << search_result[choice][0] << endl;
	cout << "1. Download Video\n2. Download Audio\n3. Play Video\n4. Play Audio\n\nCommand: ";
	int ch;
	cin >> ch;
	switch(ch){
		case 1:{
			download_video(search_result[choice][1]);
			break;
		}
		case 2:{
			download_audio(search_result[choice][1]);
			break;
		}
		case 3:{
			stream_video(search_result[choice][1]);
			break;
		}
		case 4:{
			stream_audio(search_result[choice][1]);
			break;
		}
		default:{
			search_youtube(search_command, search);
			break;
		}
	}

}

void cs(){
	system("cls");
}

void download_video(string video_url){

	title_display();
	string command = "";
	if(video_url.rfind("https://www.youtube.com/playlist?list=", 0) == 0){
		video_url = removeAll(video_url,"https://www.youtube.com/playlist?list=");
		command = "lib\\youtube-dl -o \"Downloads/Video/%(title)s.%(ext)s\" -f 136+140 https://www.youtube.com/playlist?list=" + video_url;
	}
	else{
		video_url = removeAll(video_url, "https://www.youtube.com/watch?v=");
		command = "lib\\youtube-dl -o \"Downloads/Video/%(title)s.%(ext)s\" -f 136+140 https://www.youtube.com/watch?v=" + video_url;
	}
	system(command.c_str());
	main();

}

void download_audio(string audio_url){

	title_display();
	string command = "";
	if(audio_url.rfind("https://www.youtube.com/playlist?list=", 0) == 0){
		audio_url = removeAll(audio_url, "https://www.youtube.com/playlist?list=");
		command = "lib\\youtube-dl -o \"Downloads/Audio/%(title)s.%(ext)s\"  --extract-audio --audio-format mp3 --audio-quality=320k https://www.youtube.com/playlist?list=" + audio_url;
	}
	else{
		audio_url = removeAll(audio_url, "https://www.youtube.com/watch?v=");
		command = "lib\\youtube-dl -o \"Downloads/Audio/%(title)s.%(ext)s\"  --extract-audio --audio-format mp3 --audio-quality=320k https://ww.youtube.com/watch?v=" + audio_url;
	}
	system(command.c_str());
	main();

}

void stream_video(string video_url){

	title_display();
	string command = "";
	if(video_url.rfind("https://www.youtube.com/playlist?list=", 0) == 0){
		video_url = removeAll(video_url, "https://www.youtube.com/playlist?list=");
		command = "lib\\mpv https://www.youtube.com/playlist?list=" + video_url;
	}
	else{
		video_url = removeAll(video_url, "https://www.youtube.com/watch?v=");
		command = "lib\\mpv https://www.youtube.com/watch?v=" + video_url;
	}
	system(command.c_str());
	main();

}


void stream_audio(string audio_url){

	try{
		title_display();
		string command = "";
		if(audio_url.rfind("https://www.youtube.com/playlist?list=", 0) == 0){
			audio_url = removeAll(audio_url, "https://www.youtube.com/playlist?list=");
			command = "lib\\mpv https://www.youtube.com/playlist?list="+ audio_url + " --no-video";
		}
		else{
			audio_url = removeAll(audio_url, "https://www.youtube.com/watch?v=");
			command = "lib\\mpv https://www.youtube.com/watch?v="+ audio_url + " --no-video";
		}
		system("pause");
		system(command.c_str());
		system("pause");
	}catch(int e){
		cout << "Stopped!!" << endl;
		system("pause");
		main();
	}



}

///////                                    Note
//                                      update code
//
//
//  powershell -Command "(New-Object Net.WebClient).DownloadFile('https://www.dropbox.com/s/a4cydl3y5kj29bh/version.txt?dl=1', 'version.txt')"
//  powershell -Command "Invoke-WebRequest https://www.dropbox.com/s/a4cydl3y5kj29bh/version.txt?dl=1 -OutFile version.txt"
//
//
///////

void get_update(){
	system(("lib\\updater_lib " + get_version()).c_str());
}

//void silent_update(){
//	system(("lib\\updater_lib " + get_version() + " silent").c_str());
//}

void updater_update(){
	ifstream file("lib\\updater_version.txt");
	if(file){
		string temp1, temp2;
		string updater_user_version = "";
		while(file >> temp1 >> temp2){
			updater_user_version = temp1;
		}
		file.close();
		string command1 = "powershell -Command \"(New-Object Net.WebClient).DownloadFile('https://www.dropbox.com/s/fv2jrdggmmiv4i1/updater_version.txt?dl=1', 'lib\\updater_version.txt')\"";
		string command2 = "powershell -Command \"Invoke-WebRequest https://www.dropbox.com/s/fv2jrdggmmiv4i1/updater_version.txt?dl=1 -OutFile lib\\updater_version.txt\"";
		system((command1 + "\n" + command2).c_str());
		ifstream file("lib\\updater_version.txt");
		string new_version = "";
		string version_link = "";
		title_display();
		cout << "Please wait.. ";
		if(system("lib\\youtube-dl -U")){
			get_update();
		}
		while(file >> temp1 >> temp2){
			new_version = temp1;
			version_link = temp2;
//			cout << new_version << " " << version_link << endl;
//			system("pause");
		}
		if(new_version != updater_user_version){
			title_display();
			cout << "Updating CMDYTPD Libaries.." << endl;
			system("lib\\youtube-dl -U");
			string command1 = "powershell -Command \"(New-Object Net.WebClient).DownloadFile('" + version_link + "', 'lib\\updater_lib.exe')\"";
			string command2 = "powershell -Command \"Invoke-WebRequest " + version_link + " -OutFile lib\\updater_lib.exe\"";
			system((command1 + "\n" + command2).c_str());
			title_display();
			cout << "Libraries Successfully Updated!! :)" << endl;
			system("pause");
		}
	}
	else{
		system("powershell -Command \"(New-Object Net.WebClient).DownloadFile('https://www.dropbox.com/s/tj9210a2ny56b6b/temp_updater_version.txt?dl=1', 'lib\\updater_version.txt')\"");
		system("powershell -Command \"Invoke-WebRequest https://www.dropbox.com/s/tj9210a2ny56b6b/temp_updater_version.txt?dl=1 -OutFile lib\\updater_version.txt\"");
		updater_update();
	}
}




lib/updater_lib.cpp
#include <iostream>
#include <fstream>
#include <pthread.h>
using namespace std;

/* Global Variable */
bool silent = false;


string get_version(){
	return "v1.5";
}

void *done(void *){
	system("lib\\done_update.cmd");
	return NULL;
}

void display(){
}

void kill(){
	system("taskkill /IM CMDYTPD_v3.x_beta.exe -f");
	system("taskkill /IM cmd.exe -f");
}	

main(int argc, char* argv[]){
	string command_input = "";
	system("cls");
	ifstream old_file("../CMDYTPD_v2.x_beta.exe");
	if(old_file.good()){
		remove("../CMDYTPD_v2.x_beta.exe");
		cout << "Cleaning files... " << endl;
	}
	for(int i = 1; i < argc; i++ ){
		command_input = argv[i];
	}
	cout << command_input << endl;
	if(command_input == "get-version"){
		get_version();
	}
	else{
		if(command_input == "silent"){
			silent = true;
		}
		if(!silent){
			system(("title CMDYTPD Updater " + get_version()).c_str());
			cout << "Checking The Latest Version.." << endl;
		}
		string user_version = command_input;

		try{
			string command1 = "powershell -Command \"(New-Object Net.WebClient).DownloadFile('https://www.dropbox.com/s/zl3co84h06swa2c/version.txt?dl=1', 'lib\\version.txt')\"";
			string command2 = "powershell -Command \"Invoke-WebRequest https://www.dropbox.com/s/zl3co84h06swa2c/version.txt?dl=1 -OutFile lib\\version.txt\"";
			system((command1 + "\n" + command2).c_str());
		}catch(int e){
//			system("pause");
		}
	
		ifstream file("lib\\version.txt");
		string new_version[10] = {};
		string version_link[10] = {};
		string destination[10] = {};
		string temp1, temp2, temp3;
		int i = 0;
		while(file >> temp1 >> temp2 >> temp3){
			new_version[i] = temp1;
			version_link[i] = temp2;
			destination[i] = temp3;
			i++;
			cout << new_version[i] << "-" << version_link[i] << "-" << destination[i] << endl;
		}
		file.close();
		system("cls");
		if(new_version[0] != user_version && (user_version != "silent")){
			int ch;
			while(ch!=2){
				cout << "CMDYTPD " << user_version << endl;
				cout << "HEY! THERE IS A NEW VERSION!" << endl;
				cout << "NEW VERSION: " << new_version[0] << endl;
				cout << "1. Update Now\n2. Not now\n3. ChangeLog\nCommand: " << endl;
				cin >> ch;
				switch(ch){
					case 1 :{
						system("cls");
						cout << "CMDYTPD " << user_version << endl;
						cout << "Updating.." << endl;
						int updates = 0;
						for(int i = 0; i < 10; i++){
							if(!(new_version[i].empty())){
								updates++;
							}
							else{
								break;
							}
						}
						for(int i = 0; i < updates; i++){
							system("cls");
							cout << "CMDYTPD " << user_version << endl;
							cout << "Updating.." << endl;
							cout << 100+((i-updates)*10) << "% Done.";
							string command1 = "powershell -Command \"(New-Object Net.WebClient).DownloadFile('" + version_link[i] + "', '" + destination[i] + "')\"";
							string command2 = "powershell -Command \"Invoke-WebRequest " + version_link[i] + " -OutFile " + destination[i] + "\"";
							system((command1 + "\n" + command2).c_str());
						}
						ch=2;
						cout << "UPDATE SUCCESS!" << endl;
						break;
					}
					case 2: {
						pthread_t t1;
						pthread_create(&t1, NULL, done, NULL);
						pthread_join(t1, NULL);
						break;
					}
					case 3:{
						if(!silent){
							system("cls");
							cout << "Version Changelog" << endl;
							string command1 = "powershell -Command \"(New-Object Net.WebClient).DownloadFile('https://www.dropbox.com/s/nfqtip600dfwk32/changelog.txt?dl=1', 'lib\\changelog.txt')\"";
							string command2 = "powershell -Command \"Invoke-WebRequest https://www.dropbox.com/s/nfqtip600dfwk32/changelog.txt?dl=1 -OutFile lib\\changelog.txt\"";
							system((command1 + "\n" + command2).c_str());
							ifstream file("lib\\changelog.txt");
							string line;
							while(std::getline (file,line)){
								std::string s(line);
								cout << line << endl;
							}
							file.close();
							system("pause");	
						}
						break;
					}
				}
			}
		}
		else{
			if(!silent){
				cout << "CMDYTPD " << user_version << endl;
				cout << "Your CMDYTPD is up to date! :)" << endl;
				cout << "Version Changelog" << endl;
				string command1 = "powershell -Command \"(New-Object Net.WebClient).DownloadFile('https://www.dropbox.com/s/nfqtip600dfwk32/changelog.txt?dl=1', 'lib\\changelog.txt')\"";
				string command2 = "powershell -Command \"Invoke-WebRequest https://www.dropbox.com/s/nfqtip600dfwk32/changelog.txt?dl=1 -OutFile lib\\changelog.txt\"";
				system((command1 + "\n" + command2).c_str());
				ifstream file("lib\\changelog.txt");
				string line;
				while(std::getline (file,line)){
					std::string s(line);
					cout << line << endl;
				}
				file.close();
				system("pause");
			}

		}
		if(!silent){
			pthread_t t1;
			pthread_create(&t1, NULL, done, NULL);
			pthread_join(t1, NULL);
		}
		else{
			system("exit");
		}

	}

	return 0;
}


main.cpp

/*


CMD YouTube Player and Downloader v2.1_beta

Release Update

(2.2)
1. Bug Fix

(2.1)
1. Playlist Player
2. Search History
3. Help Section Updated
4. App OTA update
5. App Optimization
6. Bug Fixed

(2.0)
1. Youtube Search added
2. App Optimization


*/
/////////






#include <iostream>
#include <conio.h>
#include "proccess_lib.h"
using namespace std;


int main(){
	system(("title CMDYTPD " + get_version()).c_str());
	updater_update();
//	silent_update();
/////////////////// Code Testing Area  ///////////////////////////////
//	cout << "Testing Code...." << endl;
//
//	ifstream file("history.txt");
//	string line = "";
//	history_count = 0;
//	while(file >> line){
//		cout << history_count << line << endl;
//		//search_history[history_count] = line;
//		//cout << search_history[history_count] << " = " << line << endl;
//		history_count++;
//	}
//
//
//	get_update();
//	cout << "\\" << endl;
//
//
//
//
//	system("pause");
////////////////////////////////////////////////////////////////////////
	int ch;
	string search = "";
	while(ch != 8){
		title_display();
		cout << "1. Search\n2. Download\n3. Play\n4. Online Radio\n5. Help\n6. About\n7. Check for Update\n8. Exit\nCommand: ";
		cin >> ch;
		switch(ch){
			case 1: {
				title_display();
				cout << "Search Query: " << endl;
				get_history(true);
				string search_command = "";
				char character = ' ';
				while(character != 13){
					if(kbhit()){
						character = getch();
						if((character == 8) && (search.length() > 0)){
							search.erase(search.length()-1);
						}
						else{
							search += character;
							string first_command = "lib\\youtube-dl ytsearch20:\"";
							string second_command = "\" --get-title --get-id";
							search_command = first_command;
							search_command += search;
							search_command += second_command;
						}
						title_display();
						cout << "Search Query: " ;
						cout << search << endl;
						get_history(true);
//						cout << search_command;
					}
				}
				cs();
				add_history(search);
				search_youtube(search_command, search);
				break;
			}
			case 2: {
				title_display();
				cout << "Download URL: " ;
				string url;
				cin >> url;
				title_display();
				cout << "1. Download Video\n2. Download Audio\n\nCommand: ";
				int ch1;
				cin >> ch1;
				switch(ch1){
					case 1:{
						download_video(url);
						break;
					}
					case 2:{
						download_audio(url);
						break;
					}
				}
				break;
			}
			case 3: {
				title_display();
				cout << "Stream URL: " ;
				string url;
				cin >> url;
				title_display();
				cout << "By Dee" << endl << endl;
				cout << "1. Stream Video\n2. Stream Audio\n\nCommand: ";
				int ch1;
				cin >> ch1;
				switch(ch1){
					case 1:{
						stream_video(url);
						break;
					}
					case 2:{
						stream_audio(url);
						break;
					}
				}
				break;
			}
			case 4: {
				get_radio();
				break;
			}
			case 5: {
				title_display();
				cout << "Help" << endl << endl;
				cout << "Key\t\t\tDescription" << endl;
				cout << "'space' or 'p'\t\tPause or Play" << endl;
				cout << "'q'\t\t\tEnd music" << endl;
				cout << "'arrow keys'\t\tSeek" << endl;
				cout << "'m'\t\t\tMute" << endl;
				cout << "'j'\t\t\tSelect Subtitle" << endl;
				cout << "'v'\t\t\tSubtitle On and Off" << endl;
				cout << "'['']'\t\t\tChange Speed" << endl;
				cout << "'9''0'\t\t\tVolume" << endl;
				cout << endl;
				system("pause");
				break;
			}
			case 6: {
				title_display();
				cout << "Copyright 2018-2019" << endl;
				cout << "\nThanks to" << endl;
				cout << "Github" << endl;
				cout << "YouTube" << endl;
				cout << "Google" << endl;
				cout << "mpv Player" << endl;
				cout << "StackOverFlow" << endl;
				cout << "" << endl;
				cout << "This App is For Personal Use Only Not For Sale" << endl;
				cout << "Download Copyright Musics at your own RISK!" << endl;
				cout << "Dont Blame me if you are in jail. :)" << endl;
				cout << "" << endl;
				cout << "About (CMDYTPD) v3.1.2_beta" << endl;
				cout << "" << endl;
				cout << "1. Added File Optimizer\n\t(3.1.2) Old files remover" << endl;
				cout << "About (CMDYTPD) v3.0_beta" << endl;
				cout << "" << endl;
				cout << "About (CMDYTPD) v3.1.1_beta" << endl;
				cout << "" << endl;
				cout << "1. Updater Improvements\n\t(3.1.1) Fixed Compatibility issue!" <<endl;
				cout << "About (CMDYTPD) v3.0_beta" << endl;
				cout << "" << endl;
				cout << "About (CMDYTPD) v3.1_beta" << endl;
				cout << "" << endl;
				cout << "1. Bug Fixes\n\t(3.1) About Section Bug Fixed!\n\t(3.1) Choosing a search result fix" << endl;
				cout << "About (CMDYTPD) v3.0_beta" << endl;
				cout << "" << endl;
				cout << "1. Optimized\n\t(3.0) Code Optimization and changes(2.2_unreleased)" << endl;
				cout << "2. Online Radio\n\t(3.0) Added Online Radio" << endl;
				cout << "3. Bug fix\n\t(3.0) OTA app update fix and improvements(2.2_unreleased)\n\t(3.0) Updater app update(2.2_unreleased)" << endl << endl;
				cout << "About (CMDYTPD) v2.1_beta" << endl;
				cout << "" << endl;
				cout << "1. Playlist\n\t(2.1) You can now Play or Download YouTube Playlists" << endl;
				cout << "2. Help Section Added\n\t(2.1) Help Section is now Updated!" << endl;
				cout << "3. Search History Added\n\t(2.1) Search History Added!" << endl;
				cout << "4. App OTA Update Added\n\t(2.1) App OTA update is now added for easy and fast application update in just one click!" << endl;
				cout << "5. Optimized!\n\t(2.1) Code improvements and optimized!" << endl;
				cout << "6. Bug Fixed!\n\t(2.1) Fixed! Search Result Bug\n\t(2.1) Fixed! Exits after playing." << endl << endl;
				cout << "About (CMDYTPD) v2.0_beta" << endl;
				cout << "" << endl;
				cout << "1. Search Menu Added!\n\t(2.0) You can now Search youtube videoes and stream or download" << endl;
				cout << "2. Optimized!\n\t(2.0) Optimized and Much Faster.." << endl;
				cout << "About (CMDYTPD) v1.0_alpha" << endl;
				cout << "" << endl;
				cout << "1. You can download and stream music on youtube\n\t(1.0) You can now download or stream music using youtube url." << endl;
				cout << "2. More Bugs To Fix Later.." << endl << endl;
				system("pause");
				break;
			}
			case 7:{
			    get_update();
				return 0;
				break;
			}
			case 8 :{
				return 0;
				break;
			}
		}

	}


}
